test/Serializer.js

```diff
@@ -0,0 +1,251 @@
+const chaiAsPromised = require("chai-as-promised");
+const chai = require("chai");
+chai.use(chaiAsPromised);
+const {expect} = chai;
+const dynamoose = require("../lib");
+const {Model} = dynamoose;
+const Serializer = require("../lib/Serializer");
+
+describe("Serializer", () => {
+	let User;
+	beforeEach(() => {
+		User = new Model("User",
+			{id: Number, name: String, email: String, phone: String, passwordHash: String, status: String},
+			{"create": false, "waitForActive": false}
+		);
+	});
+	afterEach(() => {
+		User = null;
+	});
+
+	it("Should verify Model and Document have a serializer instance and expected methods bound", () => {
+		const doc = new User({"id": 1, name: "User"});
+		expect(User.serializer).to.be.an.instanceof(Serializer);
+		expect(doc.serialize).to.be.a("function");
+		expect(User.serializeMany).to.be.a("function");
+	});
+
+	it("Should add some serializers to the Models serializer instance", () => {
+		addSerializers();
+		expect(User.serializer._serializers).to.have.property("contactInfoOnly").to.be.an("array");
+		expect(User.serializer._serializers).to.have.property("hideSecure").to.be.an("object");
+		expect(User.serializer._serializers).to.have.property("isActiveNoStatus").to.be.an("object");
+		expect(User.serializer._serializers).to.have.property("isActive").to.be.an("object");
+		expect(User.serializer._defaultSerializer).eql("_default");
+	});
+
+	it("Should remove an existing serializer from the instance and change _defaultSerializer accordingly", () => {
+		addSerializers();
+		User.serializer.remove("hideSecure");
+		expect(User.serializer._serializers).to.have.property("contactInfoOnly").to.be.an("array");
+		expect(User.serializer._serializers).to.have.property("isActiveNoStatus").to.be.an("object");
+		expect(User.serializer._serializers).to.have.property("isActive").to.be.an("object");
+		expect(User.serializer._serializers).to.not.have.property("hideSecure");
+		expect(User.serializer._defaultSerializer).eql("_default");
+
+		User.serializer.setDefault("contactInfoOnly");
+		User.serializer.setDefault("doesntExist");
+		expect(User.serializer._defaultSerializer).eql("contactInfoOnly");
+
+		User.serializer.remove("contactInfoOnly");
+		expect(User.serializer._serializers).to.not.have.property("contactInfoOnly");
+		expect(User.serializer._defaultSerializer).eql("_default");
+
+		User.serializer.remove("nonExistent");
+	});
+
+	it("Should run the document through a serializer configured with an array (include)", () => {
+		addSerializers();
+		const docs = createDocuments();
+		const result = docs[0].serialize("contactInfoOnly");
+		expect(result).to.have.property("name");
+		expect(result).to.have.property("email");
+		expect(result).to.have.property("phone");
+		expect(result).to.not.have.property("id");
+		expect(result).to.not.have.property("passwordHash");
+		expect(result).to.not.have.property("status");
```

Comment:
Something like this maybe?

```suggestion
		expect(result).to.eql(utils.object.pick(docs[0], ["name", "email", "phone"]));
```

Reply:
Will have to do it throughout.

Reply:
Done in 6a718133db0e7c471a338e8b3074938c77b50342